File: backend/workflow/roles.go
package workflow

import (
	"embed"
	"fmt"
	"strings"
	"time"
)

//go:embed roles/*.txt
var roleFiles embed.FS

func roleTemplate(role string) string {
	role = strings.ToLower(strings.TrimSpace(role))
	if role == "" {
		role = "audit"
	}
	data, err := roleFiles.ReadFile("roles/" + role + ".txt")
	if err != nil {
		return fmt.Sprintf("You are the %s subagent in a dynamic workflow. Follow the prompt exactly and return only the requested result.", role)
	}
	return strings.ReplaceAll(string(data), "{{today}}", time.Now().Format("2006-01-02"))
}

File: backend/workflow/roles.ts
// PORT: backend/workflow/roles.go

import fs from "node:fs";
import path from "node:path";
import { fileURLToPath } from "node:url";

let currentDir = "";
try {
  currentDir = path.dirname(fileURLToPath(import.meta.url));
} catch {
  currentDir = __dirname;
}

export function roleTemplate(role: string): string {
  role = role.trim().toLowerCase();
  if (role === "") {
    role = "audit";
  }
  try {
    const data = fs.readFileSync(path.join(currentDir, "roles", `${role}.txt`), "utf8");
    const now = new Date();
    const year = now.getFullYear();
    const month = String(now.getMonth() + 1).padStart(2, "0");
    const day = String(now.getDate()).padStart(2, "0");
    const today = `${year}-${month}-${day}`;
    return data.replace(/\{\{today\}\}/g, today);
  } catch {
    return `You are the ${role} subagent in a dynamic workflow. Follow the prompt exactly and return only the requested result.`;
  }
}

/*
PORT STATUS
source path: backend/workflow/roles.go
source lines: 24
draft lines: 28
confidence: high
status: phase_b_compile
*/
